bzt/modules/blazemeter.py

```diff
@@ -257,6 +269,99 @@ def set_last_status_check(self, value):
         self._last_status_check = value
         self.log.debug("Set last check time to: %s", self._last_status_check)
 
+    def monitoring_data(self, data):
+        if self.send_monitoring:
+            self.monitoring.extend(data)
+
+    def __send_monitoring(self):
+        # self.log.info("Mon: %s", self.monitoring)
+        src_name = platform.node()
+        data = self.get_monitoring_json()
+        try:
+            self.client.send_monitoring_data(src_name, data)
+        except IOError:
+            self.log.debug("Error sending data: %s", traceback.format_exc())
+            self.log.warning("Failed to send data, will retry in %s sec...", self.client.timeout)
+            try:
+                time.sleep(self.client.timeout)
+                self.client.send_monitoring_data(src_name, data)
+                self.log.info("Succeeded with retry")
+            except IOError:
+                self.log.error("Fatal error sending data: %s", traceback.format_exc())
+                self.log.warning("Will skip failed data and continue running")
+
+    def get_monitoring_json(self):
+        results = {}
+        hosts = []
+        kpis = {}
+
+        for item in self.monitoring:
+            if item['source'] == 'local':
+                item['source'] = platform.node()
+
+            src_name = item['source']
+            if src_name not in results:
+                results[src_name] = {
+                    "name": src_name,
+                    "intervals": OrderedDict()
```

Comment:
It's because of plotting behavior from BZA. It just takes JSON as we generate it and plots from it, no re-sorting done on server side. So wrong order produces wrong graph.